modules/messages/src/weights_ext.rs

```diff
@@ -410,11 +424,10 @@ pub trait WeightInfoExt: WeightInfo {
 	/// `receive_single_message_proof_with_dispatch` benchmark. See its requirements for
 	/// details.
 	fn message_dispatch_weight(message_size: u32) -> Weight {
-		// There may be a tiny overweight/underweight here, because we don't account how message
-		// size affects all steps before dispatch. But the effect should be small enough and we
-		// may ignore it.
-		Self::receive_single_message_n_bytes_proof_with_dispatch(message_size)
-			.saturating_sub(Self::receive_single_message_proof())
+		let message_size_in_kb = message_size / 1024;
+		Self::receive_single_message_n_kb_proof_with_dispatch(message_size_in_kb).saturating_sub(
+			Self::receive_single_message_n_kb_proof_with_dispatch(message_size_in_kb),
+		)
```

Comment:
:facepalm:

Reply:
Thanks for spotting that! The fix breaks this new integration test, though. I'll think if I could do something about it - will take some time

Reply:
So I had to:
1) introduce hard limit on maximal message size - it is set to `64Kb`;
2) change both benchmarks to `n_bytes` instead of `n_kb` - it gives better results and, what is more important, the cost of single byte is not zero (which was true long long time ago => had to use kilobytes)